tf_sig_build_dockerfiles/setup.python.sh

```diff
@@ -33,9 +33,17 @@ ln -sf /usr/bin/$VERSION /usr/bin/python
 ln -sf /usr/lib/$VERSION /usr/lib/tf_python
 
 # Install pip
-curl https://bootstrap.pypa.io/get-pip.py -o get-pip.py
-python3 get-pip.py
+if [[ "$VERSION" == "python3.10" ]]; then
+  # Python 3.10 pip reference is broken for pip 21.3 (https://github.com/pypa/pip/issues/10647)
+  # TODO(rameshsampath): Remove once Python 3.10 works with latest pip
+  python3 -m ensurepip
+  python3 -m pip install pip~=21.2.0
```

Comment:
Why does this need to be pinned to 21.2.x? Can you go into more detail?

Reply:
Current release is 21.3 which doesn't work with Python 3.10, so just going one minor version back.  I can add that to the text.

Reply:
I mean, what specifically doesn't work in these containers when the pip version is 21.3?

Reply:
When I install pip version 21.3, it doesn't install the packages.  Here's the problem I run into - 
```
(py310) rameshsampath@rsamdev:~/temp$ docker run -it --rm gcr.io/tensorflow-sigs/build:58-python3.10
tf-docker / > pip -V
pip 21.3.1 from /usr/lib/python3.10/dist-packages/pip (python 3.10)
tf-docker / > pip install numpy
Collecting numpy
  Downloading numpy-1.21.4-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (15.9 MB)
     |████████████████████████████████| 15.9 MB 8.4 MB/s
Installing collected packages: numpy
Successfully installed numpy-1.21.4
WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
tf-docker / > python
Python 3.10.0 (default, Oct  4 2021, 22:09:55) [GCC 9.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import numpy
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ModuleNotFoundError: No module named 'numpy'

tf-docker / > pip list
Package                 Version
----------------------- --------------------
certifi                 2019.11.28
chardet                 3.0.4
dbus-python             1.2.16
grpcio                  1.38.1
idna                    2.8
pip                     21.3.1
PyGObject               3.36.0
python-apt              2.0.0+ubuntu0.20.4.6
requests                2.22.0
requests-unixsocket     0.2.0
screen-resolution-extra 0.0.0
setuptools              57.4.0
termcolor               1.1.0
urllib3                 1.25.8
wrapt                   1.12.1
xkit                    0.0.0
```

Here it uses the latest pip 21.3, doesn't fail on requirements install, but doesn't show up in the pip list.  This is why I had to downgrade to pip 21.2.x

Reply:
Thanks! That's very helpful.

Reply:
I am hoping to come back to this in Q1 and remove this workaround.